### ISSUE TITLE
[3.0.0] Solve service startup race conditions

### ISSUE THREAD

**Issue by aldeed** [reactioncommerce/reaction#6033]
## Issue Description
In 3.0.0 there are additional Docker services, and more to come. Race conditions, both locally and during deployment, are going to crop up. We should identify and solve them.

## Possible Solution
Probably the biggest one is Hydra. API, Admin, and Storefront all try to contact Hydra admin endpoint during their startup or soon after. Some have a `wait-for` script that delays starting until Hydra is accessible, but it runs only locally so it doesn't solve the deployment problem.

I'd suggest:
- Make "wait for Hydra" code part of all three codebases
- Call it before startup and don't exit/error until a minute or so goes by without contact
- Then continue with current startup code

A similar thing could be done in services that depend on the API, but usually those do not completely stop if the API is down. Just individual requests would fail.

Soon Kafka availability will be another area to handle.

### Versions
3.0.0+

**Comment by ticean** [comment]
🐻 Rub a [cub](https://github.com/confluentinc/confluent-docker-utils/blob/master/confluent/docker_utils/cub.py), [dub](https://github.com/confluentinc/confluent-docker-utils/blob/master/confluent/docker_utils/dub.py). :smile:

**Comment by trojanh** [comment]
I would like to help with this, as far as I can understand as I am not much experienced with these scripts, we will need
- to create `bin/start` for `reaction-admin` and `reaction`, its already present for `example-storefront`
- add startup command in `bin/start` file along for with `wait-for` script like in [this file](https://github.com//reactioncommerce/example-storefront/blob/052a330e72d9658a9e69b0dedbb651c5decc10bd/bin/start#L14)
- replace docker-compose.override.yml `command` script with `command: "./bin/start"` in `reaction-admin` and `reaction`.

This should cover the hydra part at least and could be reused for other services as well, right?

**Comment by spencern** [comment]
Linking this issue: https://github.com/reactioncommerce/reaction-platform/issues/89

**Comment by mpaktiti** [comment]
I recently updated my local environment to v3. The first time I started up `reaction-admin` and `reaction-identity` I got errors from both that they couldn’t connect to mongo (which was up and running, I could connect via command line). I took both these containers down and started them up again, and got no more errors. It hasn't happened since, only that one time, and I don't know if it's an actual problem or something that happened only in my environment. I'm leaving this comment here in any case.
